app/src/main/java/xyz/jhughes/laundry/fragments/MachineFragment.java

```diff
@@ -187,43 +181,34 @@ private void alertNetworkError() {
     }
 
     private void removeNotifyOnAvailableButton() {
-        if (layout == null) return;
-        layout.removeView(notifyButton);
-        notifyButtonVisible = false;
+        notifyButton.setVisibility(View.GONE);
     }
 
     private void addNotifyOnAvailableButton() {
-        if (layout == null) return;
-        layout.addView(notifyButton, 0);
-        notifyButtonVisible = true;
+        notifyButton.setVisibility(View.VISIBLE);
     }
 
     private void initializeNotifyOnAvaiableButton() {
         final String text = isDryers ? getString(R.string.notify_on_dryer_available) : getString(R.string.notify_on_washer_available);
-        layout = (LinearLayout) rootView.findViewById(R.id.machine_list_wrapper);
-        notifyButton = new Button(getContext());
+        notifyButton.setText(text);
         notifyButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
                 Machine m = null;
                 int mTime = Integer.MAX_VALUE;
                 for (Machine mach : currentAdapter.getCurrentMachines()) {
-                    if (m == null) {
-                        m = mach;
-                    } else {
-                        try {
-                            int machTime = Integer.parseInt(mach.getTime().substring(0, mach.getTime().indexOf(' ')));
-                            if (machTime < mTime) {
-                                m = mach;
-                                mTime = machTime;
-                            }
-                        } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
-                            continue;
+                    try {
+                        int machTime = Integer.parseInt(mach.getTime().substring(0, mach.getTime().indexOf(' ')));
+                        if (machTime < mTime) {
+                            m = mach;
+                            mTime = machTime;
                         }
+                    } catch (NumberFormatException | StringIndexOutOfBoundsException e) {
+                        continue;
                     }
                 }
                 if (m == null) {
-                    postSnackbar("No machines to set timer for!", Snackbar.LENGTH_LONG);
+                    postSnackbar("No machines to set a timer for!", Snackbar.LENGTH_LONG);
```

Comment:
Could we move this to strings.xml? I know it wasn't there to begin with. Blame Pitlor.